File: hormigas.py


alambre = []
centro = float

largo = raw_input("ingrese largo del alambre: ")
n_hormigas = raw_input("ingrese la cantidad de hormigas: ")

while(int(n_hormigas) > int(largo)):
    print "largo = ",largo
    print "hormigas = ",n_hormigas
    n_hormigas = raw_input("el numero de hormigas no puede superar el largo del alambre.\n reingresar numero de hormigas: ")

centro = float(largo)/2
print "centro del alambre ", centro

for i in range(1,int(largo) + 1):
    alambre.append(int(i))

print "Largo del alambre", alambre

print"Hay en total", n_hormigas,"hormigas en un alambre de",largo,"centimetros. "


for i in range(1,int(n_hormigas)+1):
    print"poscion de la hormiga ",i, " : "
    pocision = raw_input()

    while int(pocision) > int(largo):
        pocision= raw_input("la ormiga esta afuera del alambre, reingresar pocision: ")

    while "hormiga" == str(alambre[int(pocision)-1]):
        pocision= raw_input("Ya hay una hormiga aqui, reingresar pocision: ")

    alambre[int(pocision)-1] = "hormiga"

print alambre

File: hormigas.c
#include<stdio.h>
#include<stdlib.h>


typedef struct
  {
     int numero;
     int pos_inicial;
     int pos_actual;
     int direccion;

  } hormiga;

  void mostrar_hormigas(hormiga array[], int num)
  {
    for(int i = 0; i<num;i++)
    {
      printf("hormiga numero %d con pocision en %d con direccion %d que partio en %d \n", array[i].numero, array[i].pos_actual, array[i].direccion ,array[i].pos_inicial );
    }
  }

int main(int argc, char *argv[])
{
    int n_hormigas = 0;

    printf("ingrese largo del alambre ")
    scanf()

    printf("ingrese numero de hormigas");
    scanf("%d",&n_hormigas);

    hormiga hormigas[n_hormigas];

    int pos;
    for(int i = 0; i <n_hormigas; i++)
    {
        hormigas[i].numero = i+1;
        printf("ingrese pocision de la hormiga %d \n", i+1);
        scanf("%d", &pos);
        hormigas[i].pos_inicial = pos;
        hormigas[i].pos_actual = pos;
        hormigas[i].direccion = 0;
    }

    mostrar_hormigas(hormigas,n_hormigas);

}
